relay-genesis/update-mock-relay.js

```diff
@@ -15,9 +15,9 @@ module.exports = async function() {
 
         const mockRelay = await MockRelay.deployed()
 
-        await mockRelay.setCurrentEpochDifficulty(difficulty)
-        await mockRelay.setPrevEpochDifficulty(difficulty)
+        await mockRelay.setCurrentEpochDifficulty(Math.min(difficulty, 1))
+        await mockRelay.setPrevEpochDifficulty(Math.min(difficulty, 1))
```

Comment:
Why are we changing this? The `MockRelay` contract has `1` as default current and previous difficulty. IIRC, the regtest network has a difficulty less than `1` so actually this script sets the values in the contract to `0` and due to that, allows transaction proofs to be accepted successfully. Am I missing something?

Reply:
This should definitely be Math.max, not min. My goal here was to make sure running this did not break the mock relay if someone set it while running regtest. Does that make sense?

Reply:
Hmm, but if we use `Math.max` we can never set the difficulty to a value less than `1` which is needed when we use`regtest`.

Reply:
regtest has a difficulty less than 1? 🤔 The contracts return a minimum difficulty of 0 for SPV proofs, so I don't think we can verify a 0-difficulty transaction. I might be misunderstanding a few bits here.

Reply:
The [`getdifficulty`](https://bitcoincore.org/en/doc/0.20.0/rpc/blockchain/getdifficulty/) method of the bitcoin RPC returns something between `0` and `1` when executed against a `regtest` client. But, solidity just cuts the decimal part and we have a `0` set in the `MockRelay` contract.

Reply:
Mmkay, so I'm just going to revert this for now and we'll see if I can reproduce my prior issues.